miner/worker.go

```diff
@@ -785,25 +786,46 @@ func (w *worker) updateSnapshot() {
 }
 
 func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Address) ([]*types.Log, error) {
-	snap := w.current.state.Snapshot()
-	privateSnap := w.current.privateState.Snapshot()
-
+	workerEnv := w.current
```

Comment:
would make sense to encapsulate this block of code before the `core.ApplyTransaction(..)` into a different quorum specific function?